tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/FederationEvalStrategy.java

```diff
@@ -516,8 +519,23 @@ protected void optimizeJoinOrder(TupleExpr query, QueryInfo queryInfo, GenericIn
 	public CloseableIteration<BindingSet, QueryEvaluationException> evaluateService(FedXService service,
 			final List<BindingSet> bindings) throws QueryEvaluationException {
 
-		return new ServiceJoinIterator(new CollectionIteration<>(bindings),
-				service.getService(), EmptyBindingSet.getInstance(), this);
+		Var serviceRef = service.getService().getServiceRef();
+		String serviceUri;
+		if (serviceRef.hasValue())
+			serviceUri = serviceRef.getValue().stringValue();
```

Comment:
This is nitpicky but we have a style agreement to always use curly braces for if-then-else clauses, even if they're single line.

Reply:
I fully agree, must have missed this. Will amend the change and add the brackets. Thanks

Reply:
Added this to the first commit. No other changes added